src/aws/cloudwatch-sink.cpp

```diff
@@ -377,11 +378,18 @@ void CloudWatchSink::dump(Log const& log, Channel const& channel, Logger::Contex
     }
     try
     {
+        std::unique_lock<std::mutex> lock(logs_mtx_.get());
         Aws::CloudWatchLogs::Model::InputLogEvent e;
+        auto message = formatted_json(log, channel, context_info);
+        auto log_name = log_stream_name(log, channel);
+        if (message.empty() || log_name.empty())
+        {
+            return;
+        }
```

Comment:
any reason these need to be under a lock?

Reply:
the coredump had two issues, one with weird empty logs and the other with concurrency, so i added a lock here